Test/civl/inductive-sequentialization/PingPong.bpl

```diff
@@ -1,16 +1,32 @@
 // RUN: %parallel-boogie "%s" > "%t"
 // RUN: %diff "%s.expect" "%t"
 
-type {:linear "pid"} Pid = int;
-var {:layer 0,3} ping_channel:[int]int; // Incoming channel of Ping
-var {:layer 0,3} pong_channel:[int]int; // Incoming channel of Pong
+// This example shows how to use a bidirectional shared channel to communicate between two processes.
+
+// A bidirectional channel is a pair of ordinary channels with two ends---left and right.
+type {:datatype} ChannelPair;
+function {:constructor} ChannelPair(left: [int]int, right: [int]int): ChannelPair;
+
+type ChannelId;
+var {:layer 0,3} channel: [ChannelId]ChannelPair;
+
+// The id of a bidirectional channel can be split into two permissions---Left and Right.
+// Left permission is used to receive from the left channel and send to the right channel.
+// Right permission is used to receive from the right channel and send to the left channel.
+type {:linear "cid"} {:datatype} Permission;
+function {:constructor} Left(cid: ChannelId): Permission;
+function {:constructor} Right(cid: ChannelId): Permission;
+function {:inline} ChannelId(p: Permission) : ChannelId {
+  if is#Left(p) then cid#Left(p) else cid#Right(p)
+}
 
-const unique ping_id:int;
-const unique pong_id:int;
+function {:inline} {:linear "cid"} ChannelIdCollector(cid: ChannelId) : [Permission]bool {
+  MapConst(false)[Left(cid) := true][Right(cid) := true]
+}
 
 type {:pending_async}{:datatype} PA;
-function {:constructor} PING(x:int, pid:int) : PA;
-function {:constructor} PONG(x:int, pid:int) : PA;
+function {:constructor} PING(x: int, left: Permission): PA;
+function {:constructor} PONG(x: int, right: Permission): PA;
```

Comment:
Maybe instead of `left` and `right` we should use the same generic name for both. Like `permission` or `channel_handle`. Same in the corresponding action definitions.

On the one hand the names provide hints about the intended values, but on the other hand they could be misleading to assume something about the values before we prove it (like `assert is#Left(left);` in `PING`).

Reply:
I left the field names as they are.  But I added a comment.